### PR TITLE
POM update

### PR DESCRIPTION
**Pull request by jglick**
@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,6 +69,21 @@
         <artifactId>scm-api</artifactId>
         <version>${scm-api.version}</version>
       </dependency>
+      <dependency>
```
Comment (svanoort) [inline pom.xml]: @jglick Why were these added?  I don't see any new code that would demand them?

Comment (casz) [reply pom.xml]: upperbound dependency requirement 😭

Comment (svanoort) [reply pom.xml]: Sigh, those buggers.  Okay, well I think for this is really needs to explain that in a comment @jglick because otherwise this will generate head-scratching since they're not actually *used* here.

Comment (jglick) [reply pom.xml]: > they're not actually _used_ here

Which is why they are in `dependencyManagement`, not `dependencies`.

My plan is to implement [JENKINS-47498](https://issues.jenkins-ci.org/browse/JENKINS-47498), after which we could just update the parent and delete a section like this.

**Comment by jglick** [comment]
> a Java 8 update

This would require a baseline update to 2.60.x or higher. No problem from my PoV but up to the plugin maintainer to decide that. Here I am just making sure the plugin has a modern enough POM structure that it is possible to test things like https://github.com/jenkinsci/jenkins/pull/3120 against it easily. (The Mockito craziness in the tests here caused some issues.)

**Comment by jglick** [comment]
Anyway, @reviewbybees done (not sure who maintains this).

**Comment by casz** [comment]
I think that is another issue.

Nobody seems is maintaining the project and PRs are suffering for it.
Just to list a few that is severe #79 #84 #86 

I would like to step up as co-maintainer if that is okay with the bees 🐝 that used to maintain the project. @stephenc and @amuniz

**Comment by stephenc** [comment]
Well @i386 is supposed to have assigned shepherds for this plugin... I'm only stepping in on architectural issues

**Comment by casz** [comment]
Is he not already pretty busy especially with Jenkins tutorial and all the other plugins and blue ocean work, he does 😄

**Comment by i386** [comment]
@rsandell is our new Shepard for this component :)

**Comment by rsandell** [comment]
I just had to convince the bot to make me a committer :)
